lib/Service/Provisioning/Config.php

```diff
@@ -122,6 +122,9 @@ private function buildUserEmail(string $original, IUser $user) {
 		if ($user->getUID() !== null) {
 			$original = str_replace('%USERID%', $user->getUID(), $original);
 		}
+		if ($user->getDisplayName() !== null) {
+			$original = str_replace('%DISPLAYNAME%', $user->getDisplayName(), $original);
```

Comment:
Please add to https://github.com/nextcloud/mail/blob/df939b93e2993749fbdb43401acedb7cde1adf00/src/components/settings/ProvisioningSettings.vue#L246 as well